src/OrchardCore/OrchardCore.DisplayManagement.Liquid/Values/StringValuesValue.cs

```diff
@@ -0,0 +1,230 @@
+using System.Globalization;
+using System.Text.Encodings.Web;
+using Fluid;
+using Fluid.Values;
+using Microsoft.Extensions.Primitives;
+
+namespace OrchardCore.DisplayManagement.Liquid.Values;
+
+internal sealed class StringValuesValue : FluidValue
+{
+    private readonly StringValues _stringValues;
+
+    public StringValuesValue(StringValues stringValues)
+    {
+        _stringValues = stringValues;
+    }
+
+    public override FluidValues Type => FluidValues.Array;
+
+    public override bool Equals(object obj)
+    {
+        return Equals(obj as FluidValue);
+    }
+
+    public override bool Equals(FluidValue other)
+    {
+        if (other is null)
+        {
+            return _stringValues.Count == 0;
+        }
```

Comment:
Shouldn't this rather be `false`, since we're comparing a `null` object to a non-`null` one? I don't think equating a `null` and empty collection is correct.

Reply:
Yes, good catch.